lib/util/SparqlExpressionEvaluator.js

```diff
@@ -1,17 +1,44 @@
-/*! @license MIT ©2014-2016 Ruben Verborgh, Ghent University - imec */
+/*! @license MIT ©2014-2016 Ruben Verborgh, Miel Vander Sande, Ghent University - imec */
 
 var N3Util = require('n3').Util,
-    createErrorType = require('./CustomError');
+    UUID = require('uuid'),
+    crypto = require('cryptojs');
 
 var XSD = 'http://www.w3.org/2001/XMLSchema#',
-    XSD_INTEGER = XSD + 'integer',
-    XSD_DOUBLE  = XSD + 'double',
     XSD_BOOLEAN = XSD + 'boolean',
-    XSD_TRUE  = '"true"^^'  + XSD_BOOLEAN,
+    XSD_STRING = XSD + 'string',
+    XSD_DATETIME = XSD + 'dateTime',
+    XSD_TRUE = '"true"^^' + XSD_BOOLEAN,
     XSD_FALSE = '"false"^^' + XSD_BOOLEAN;
 
-var evaluators, operators,
-    UnsupportedExpressionError, UnsupportedOperatorError, InvalidArgumentsNumberError;
+var operators, evaluators;
+
+// Add all numeric values
+var numericTypes = [
+  'integer',
+  'decimal',
+  'float',
+  'double',
+  'nonPositiveInteger',
+  'negativeInteger',
+  'long',
+  'int',
+  'short',
+  'byte',
+  'nonNegativeInteger',
+  'unsignedLong',
+  'unsignedInt',
+  'unsignedShort',
+  'unsignedByte',
+  'positiveInteger',
+].map(function (type) {
+  return XSD + type;
+});
```

Comment:
TODO This should be a hash, not an array, since it is only used for membership checks (`indexOf` has linear time, `in` has constant time).